tsdb/index/index.go

```diff
@@ -864,7 +864,10 @@ func (w *Writer) writePostingsToTmpFiles() error {
 		// using more memory than a single label name can.
 		for len(names) > 0 {
 			if w.labelNames[names[0]]+c > maxPostings {
-				break
+				if len(batchNames) > 0 {
```

Comment:
It may be better to put `if c > 0` here.  This should be equivalent - they are both added to in the same place - but it makes it clearer that `w.labelNames[names[0]] > maxPostings`, which matches the "corruption" message.

Reply:
Makes sense. Pushed the change.